### PR TITLE
Add optional prefix for annotations

### PR DESCRIPTION
**Pull request by antonok-edm**
There were a few people interested in issue #16. I just stumbled upon this crate -- I love the idea behind it, but I also need that feature to use it in my own work since I have a few different structs with methods of the same name.

I think having automatic full path names would be even nicer than this solution, but I'm not that well versed with writing language extensions just yet. In the meantime, this works well enough for my (and hopefully others') needs.

### CODE REVIEW

File: tests/opt_name.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+#[noflame]
+fn c() {
+    b()
+}
+
+pub struct Lower {
+    
+}
```
Comment (llogiq) [inline tests/opt_name.rs]: Is there a reason not to write this as `pub struct Lower;`?

File: src/lib.rs [hunk 2]
```diff
@@ -15,22 +15,40 @@ use syntax::feature_gate::AttributeType;
 use syntax::symbol::Symbol;
 use syntax::util::small_vector::SmallVector;
 
-pub fn insert_flame_guard(cx: &mut ExtCtxt, _span: Span, _mi: &MetaItem,
+pub fn insert_flame_guard(cx: &mut ExtCtxt, _span: Span, mi: &MetaItem,
                           a: Annotatable) -> Annotatable {
+    let opt_ident = match mi.node {
+        MetaItemKind::Word => None,
+        MetaItemKind::List(ref v) => {
```
Comment (llogiq) [inline src/lib.rs]: Could this be simplified to match against `MetaItemKind::List(ref v) if v.len() == 1 => { .. }` and `_ => None`?

**Comment by antonok-edm** [comment]
Thanks, I didn't know about those tricks back when I wrote this :)

I also fixed the merge conflict that arised from removing quote_block.

Should be all set now.

**Comment by llogiq** [comment]
Thank you!

**Pull request merged**